### PR TITLE
Add -Zunpretty=stable-mir output test

### PR DESCRIPTION
**Pull request by ouz-a**
As strongly suggested here https://github.com/rust-lang/rust/pull/118364#issuecomment-1827974148 this adds output test for `-Zunpretty=stable-mir`, added test shows almost all the functionality of the current printer. 

r? @compiler-errors

### CODE REVIEW

**Comment by compiler-errors** [comment]
r=me when green

**Comment by compiler-errors** [comment]
This caused a test failure in https://github.com/rust-lang/rust/pull/118380#issuecomment-1828481553

@bors r-

**Comment by ouz-a** [comment]
I don't know why test failed but added `-Z mir-opt-level=3` and removed `todo!` from pretty printer

File: compiler/stable_mir/src/mir/pretty.rs
Comment (celinval) [inline compiler/stable_mir/src/mir/pretty.rs:61]: Can we print some sort of TODO message for now instead of silently ignoring the statement?

File: tests/ui/stable-mir-print/basic_function.stdout
Comment (celinval) [inline tests/ui/stable-mir-print/basic_function.stdout:17]: Not for this PR, but we probably don't want to print the structure itself, but what it represents.

Comment (celinval) [reply tests/ui/stable-mir-print/basic_function.stdout:17]: Actually I just noticed that you fixed that in #118364. :smile:

Comment (compiler-errors) [reply tests/ui/stable-mir-print/basic_function.stdout:17]: Yeah, I wanted this PR just to print the status quo, and subsequent PRs could actually do the real output fixing.

**Comment by celinval** [comment]
Hey @ouz-a, I was wondering what's blocking you in this PR.

**Comment by ouz-a** [comment]
> Hey @ouz-a, I was wondering what's blocking you in this PR.

I don't think anything is blocking this pr after rebase it should be ready to merge

File: compiler/stable_mir/src/mir/pretty.rs
Comment (celinval) [inline compiler/stable_mir/src/mir/pretty.rs:70]: Can you at least add a fixme comment to finish these

**Review by celinval — APPROVED** [review]
r=me after you add a comment and resolved the conflicts.

File: compiler/stable_mir/src/mir/pretty.rs
Comment (ouz-a) [reply compiler/stable_mir/src/mir/pretty.rs:61]: Now these return `StatementKind::X:Unimplemented`

File: compiler/stable_mir/src/mir/pretty.rs
Comment (ouz-a) [reply compiler/stable_mir/src/mir/pretty.rs]: Added fix me and now they return string explaining they are unimplemented

File: compiler/stable_mir/src/mir/pretty.rs
Comment (matthiaskrgr) [inline compiler/stable_mir/src/mir/pretty.rs:85]: these can just be `String::from`s or `to_string()`s instead of macros btw :sweat_smile:

**Comment by ouz-a** [comment]
Added flag to test so it only runs on `x86_64` so hopefully it doesn't break this time.

**Pull request merged**
